### PR TITLE
Make port, jvm memory size and threads configurable at container startup

### PR DESCRIPTION
**Pull request by sychan**
Also move default glassfish http listener port off 8080 so that workspace service can use that port if necessary (which we want in order to make container deployments consistent)

### CODE REVIEW

File: deployment/conf/.templates/start_workspace.sh.templ [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env sh
```
Comment (MrCreosote) [inline deployment/conf/.templates/start_workspace.sh.templ]: Delete the old start file

File: server_scripts/glassfish_administer_service.py [hunk 2]
```diff
@@ -76,6 +76,7 @@ def __init__(self, asadminpath, domain, domainpath, verbose):
         else:
             print('Creating domain ' + self.domain + p)
             print(self._run_local_command('create-domain', '--nopassword=true',
+                                          '--instanceport=32768',  # move instanceport off 8080
```
Comment (MrCreosote) [inline server_scripts/glassfish_administer_service.py]: This should be a cli parameter

File: deployment/conf/.templates/start_workspace.sh.templ (hunk 1 above)
Comment (sychan) [reply deployment/conf/.templates/start_workspace.sh.templ]: Done.

File: server_scripts/glassfish_administer_service.py (hunk 2 above)
Comment (sychan) [reply server_scripts/glassfish_administer_service.py]: Done.

**Comment by sychan** [comment]
@MrCreosote Changes made

File: deployment/conf/.templates/start_workspace.sh.templ [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+cd /kb/deployment
+
+# Values for port, threads, min/max memory are populated into source template from environment vars
+# the defaults are based on
+bin/glassfish_administer_service.py --verbose --admin $GLASSFISH/bin/asadmin --domain Workspace \
```
Comment (MrCreosote) [inline deployment/conf/.templates/start_workspace.sh.templ]: I assume you want an --instanceport argument here

File: server_scripts/glassfish_administer_service.py [hunk 4]
```diff
@@ -55,11 +57,12 @@ def _parseArgs():
 
 class CommandGlassfishDomain(object):
 
-    def __init__(self, asadminpath, domain, domainpath, verbose):
+    def __init__(self, asadminpath, domain, domainpath, verbose, instanceport):
         self.asadminpath = asadminpath
         self.domain = domain
         self.path = None
         self.verbose = verbose
+        self.instanceport = instanceport
```
Comment (MrCreosote) [inline server_scripts/glassfish_administer_service.py]: I don't think it needs to be an instance variable, does it? It's only used in the constructor.

File: server_scripts/glassfish_administer_service.py [hunk 5]
```diff
@@ -76,6 +79,7 @@ def __init__(self, asadminpath, domain, domainpath, verbose):
         else:
             print('Creating domain ' + self.domain + p)
             print(self._run_local_command('create-domain', '--nopassword=true',
+                                          '--instanceport='+str(self.instanceport),  # move instanceport off 8080
```
Comment (MrCreosote) [inline server_scripts/glassfish_administer_service.py]: add spaces around operators

File: server_scripts/glassfish_administer_service.py [hunk 6]
```diff
@@ -35,6 +35,8 @@ def _parseArgs():
                         'glassfish/domains.')
     parser.add_argument('-p', '--port', required=True, type=int,
                         help='the port where the application runs.')
+    parser.add_argument('-i', '--instanceport', type=int, default=8080,
+                        help='port for glassfish -instanceport option in create-domain')
```
Comment (MrCreosote) [inline server_scripts/glassfish_administer_service.py]: missing hyphen

File: deployment/conf/.templates/start_workspace.sh.templ (hunk 3 above)
Comment (sychan) [reply deployment/conf/.templates/start_workspace.sh.templ]: Not really, there is a default in the glassfish admin script that just uses the normal port 8080, which is the hard coded default in glassfish.

Comment (sychan) [reply deployment/conf/.templates/start_workspace.sh.templ]: Okay, forgot to commit that update

File: server_scripts/glassfish_administer_service.py (hunk 6 above)
Comment (sychan) [reply server_scripts/glassfish_administer_service.py]: fixed

File: server_scripts/glassfish_administer_service.py (hunk 4 above)
Comment (sychan) [reply server_scripts/glassfish_administer_service.py]: fixed.

File: server_scripts/glassfish_administer_service.py (hunk 5 above)
Comment (sychan) [reply server_scripts/glassfish_administer_service.py]: fixed.

**Pull request merged**